### PR TITLE
Support nested enum type parsing.

### PR DESCRIPTION
**Pull request by shazeline**
the main code changes are in schema.py L376-384 and L586-597 

the rest iswhitespace stripping that my editor did automatically

### CODE REVIEW

**Comment by clin7** [comment]
What is the nested enum type? Avro enum cannot be nested.

File: lang/py/src/avro/schema.py [hunk 1]
```diff
@@ -181,19 +181,19 @@ def __init__(self, name_attr, space_attr, default_space):
...
       fail_msg = 'Default space must be non-empty string or None.'
       raise SchemaParseException(fail_msg)
     elif name_attr == "":
       fail_msg = 'Default must be non-empty string or None.'
       raise SchemaParseException(fail_msg)
-    
-    self._full = None; 
-    
+
+    self._full = None;
```
Comment (premsantosh) [inline lang/py/src/avro/schema.py:192]: Can't believe this code has semi-colons :person_frowning:

**Comment by shazeline** [comment]
The current implementation only parses the avro enum field itself but not instances where the enum field is part of a record schema. 

This change fixes that.

**Comment by premsantosh** [comment]
:+1: :shipit: But in the future I'd recommend that you only commit your changes without the formatting changes for the review. It would make it easier to review the actual change :smile:  You can including the formatting changes before merging.

**Comment by clin7** [comment]
It should, the `make_avsc_object` is a recursive call.

```
>>> import avro
>>> sch = {"type": "record", "name": "foo", "fields": [{"name": "bar", "type": {"type": "enum", "name": "baz", "symbols": ["a"]}}]}
>>> from avro import schema
>>> schema.make_avsc_object(sch)
<avro.schema.RecordSchema object at 0x7f5ef2f22bd0>
>>> obj = schema.make_avsc_object(sch)
>>> obj.fields[0].type
<avro.schema.EnumSchema object at 0x7f5ef2f22d90>
```

**Comment by shazeline** [comment]
@clin7 I realized that I was constructing enum fields incorrectly

this is invalid
`{u'symbols': ["a"], 'type': 'enum', 'name': 'baz'}`

this is valid
`{"name": "bar", "type": {"type": "enum", "name": "baz", "symbols": ["a"]}}`

Thanks! I'm going to close this PR.

**Pull request closed without merging**